### PR TITLE
Compute Sentiment Analysis on Journal Text

### PR DESCRIPTION
**Pull request by mariah-crawford**
Computes sentiment analysis on journal entry text and converts the score of sentiment to a range between 1 and 10 to correspond with emoji mapping.

### CODE REVIEW

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ function addRandomQuote() {
  */
 function addPastEntries() {
   fetch('/my-data-url').then(response => response.json()).then((pastEntries) => {
+    console.log(pastEntries);
```
Comment (carolinar7) [inline src/main/webapp/script.js]: Remove this log

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -98,4 +108,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // Redirect back to the HTML page.
     response.sendRedirect("/index.html");
   }
+
+  /** Identifies the sentiment in the journal text entry string. */
```
Comment (carolinar7) [inline src/main/java/com/google/sps/servlets/DataServlet.java]: Not a big deal, but keep consistency of code by using // for single line comments

**Review by carolinar7 — CHANGES_REQUESTED** [review]
Looks good, just change a few lines. I had issues on my end...

image
image

I got a PERMISSION_DENIED error when I tried running it. I'm not sure how that can be fixed.

Also I made a pull request to add the lyrics on the server side, so that you can do sentiment analysis on the lyrics too. In that case you can find the average moodScale of the textEntry and the lyrics.

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -98,4 +108,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+        throw new Exception("Exception: Sentiment is null");
+      }
+      // Convert score ranges from -1.0:1.0 to 1:10 to be compatible with emoji mapping
+      double score = sentiment.getScore() + 1;
+      double oldRange = 2.0;
+      double newRange = 9.0;
+      double newValue = (((score) * newRange) / oldRange) + 1.0;
+      int newScore = (int)Math.round(newValue);
+      return newScore;
+    } catch (Exception e) {
```
Comment (dprado981) [inline src/main/java/com/google/sps/servlets/DataServlet.java]: Are there any exceptions that could be thrown (that aren't preventable), other than the one that's thrown if the sentiment is null, that could be thrown here? If so try to catch the specific exceptions in their own catch block. If there aren't any other exceptions, you could limit the try block to the relevant section or return -1 at the if (sentiment == null) statement

Comment (dprado981) [reply src/main/java/com/google/sps/servlets/DataServlet.java]: Another solution, if the try-catch block is needed around the entire method could be to make the method throw an exception using the throws keyword, and catch the exception when you call it

**Review by dprado981 — CHANGES_REQUESTED** [review]
Looks pretty good to me! Just double check the exception throwing/catching in analyzeSentimentText()

**Comment by carolinar7** [comment]
`git pull` before making changes

File: src/main/webapp/script.js (hunk 1 above)
Comment (mariah-crawford) [reply src/main/webapp/script.js]: Done :)

File: src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (mariah-crawford) [reply src/main/java/com/google/sps/servlets/DataServlet.java]: Done :)

File: src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (mariah-crawford) [reply src/main/java/com/google/sps/servlets/DataServlet.java]: I used the try and catch within the doPost method instead

**Comment by mariah-crawford** [comment]
> Looks good, just change a few lines. I had issues on my end...
> 
> image
> image
> 
> I got a PERMISSION_DENIED error when I tried running it. I'm not sure how that can be fixed.
> 
> Also I made a pull request to add the lyrics on the server side, so that you can do sentiment analysis on the lyrics too. In that case you can find the average moodScale of the textEntry and the lyrics.

In order to run the site using the sentiment analysis api locally on your computer, you can create a service account and follow the following instructions: [here](https://cloud.google.com/docs/authentication/getting-started).

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -94,19 +104,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
     // Ensure that form is filled out before saving to datastore
     if (textEntryString != null && !textEntryString.isEmpty()) {
-      // Convert the mood input to an int.
-      int moodScale = Integer.parseInt(moodScaleString);
+      // Get Sentiment Analysis of Journal Entry
+      int moodScale = 0;
+      try{
+        moodScale = analyzeSentimentText(textEntryString);
+      }catch (Exception e) {
```
Comment (dprado981) [inline src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: add space after try and before catch

**Review by dprado981 — APPROVED** [review]
Looks good, thanks for making the changes!

File: src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (mariah-crawford) [reply src/main/java/com/google/sps/servlets/DataServlet.java]: Done :)

**Review by carolinar7 — APPROVED** [review]
Looks good

**Pull request merged**